packages/build-tools/src/builders/ios.ts

```diff
@@ -170,9 +215,49 @@ async function buildAsync(ctx: BuildContext<Ios.Job>): Promise<void> {
     });
   });
 
-  await ctx.runBuildPhase(BuildPhase.SAVE_CACHE, async () => {
-    await ctx.cacheManager?.saveCache(ctx);
-  });
+  if (ctx.env.EAS_BUILD_RUNNER === 'eas-build') {
```

Comment:
Let's have the build phase run and skip the implementation if EAS_BUILD_RUNNER !== eas-build. This will also deindent the code because you can early return instead of wrapping big chunk of code with `if`